gst/nnstreamer/tensor_query/tensor_query_serversrc.c

```diff
@@ -14,105 +14,319 @@
+
+  gst_tensors_config_from_peer (bsrc->srcpad, &config, NULL);
+  gst_tensors_info_copy (&src->src_info, &config.info);
+  gst_tensors_config_free (&config);
+
+  if (!src->server_data) {
+    nns_loge ("Server_data is NULL");
+    return FALSE;
+  }
+
+  if (nnstreamer_query_server_data_setup (src->server_data, src->protocol,
+          src->host, src->port) != 0) {
+    nns_loge ("Failed to setup server");
+    return FALSE;
+  }
+
+  return TRUE;
+}
+
+/**
+ * @brief stop processing of query_serversrc
+ */
+static gboolean
+gst_tensor_query_serversrc_stop (GstBaseSrc * bsrc)
+{
+  GstTensorQueryServerSrc *src = GST_TENSOR_QUERY_SERVERSRC (bsrc);
+  nnstreamer_query_server_data_free (src->server_data);
+  return TRUE;
 }
 
-#pragma GCC diagnostic pop
+/**
+ * @brief create query_serversrc, wait on socket and receive data
+ */
+static GstFlowReturn
+gst_tensor_query_serversrc_create (GstBaseSrc * bsrc, guint64 offset,
+    guint size, GstBuffer ** outbuf)
+{
+  GstTensorQueryServerSrc *src = GST_TENSOR_QUERY_SERVERSRC (bsrc);
+  GstMemory *mem = NULL;
+  GstMapInfo map;
+  GstMetaQuery *meta_query;
+  TensorQueryCommandData data;
+  gsize data_size;
+  gint64 client_id;
+
+  if (!src->server_data) {
+    nns_loge ("Server data is NULL");
+    return GST_FLOW_ERROR;
+  }
+
+  client_id = nnstreamer_query_server_accept (src->server_data);
+  if (client_id == 0) {
+    nns_loge ("Failed to accept client");
+    return GST_FLOW_ERROR;
+  }
+
+  /* serve query */
+  while (1) {
+    if (nnstreamer_query_receive (client_id, &data, src->timeout) != 0) {
+      nns_loge ("Failed to receive data");
+      return GST_FLOW_ERROR;
+    }
+
+    switch (data.cmd) {
+      case _TENSOR_QUERY_CMD_TRANSFER_START:
+        /* validate tensor info */
+        if (!gst_tensors_info_is_equal (&data.data_info.config.info,
+                &src->src_info)) {
+          data.cmd = _TENSOR_QUERY_CMD_TRANSFER_DENY;
+          nns_loge ("Tensor info is not compatible");
+          if (nnstreamer_query_send (client_id, &data, sizeof (data)) != 0)
+            nns_loge ("Failed to send denial");
+          return GST_FLOW_ERROR;
+        }
+        /* send ack msg */
+        data.cmd = _TENSOR_QUERY_CMD_TRANSFER_APPROVE;
+        if (nnstreamer_query_send (client_id, &data, sizeof (data)) != 0) {
+          nns_loge ("Failed to send approval");
+          return GST_FLOW_ERROR;
+        }
+
+        /* prepare buffer */
+        data_size = gst_tensors_info_get_size (&src->src_info, -1);
+        *outbuf = gst_buffer_new_allocate (NULL, data_size, NULL);
+        mem = gst_buffer_get_all_memory (*outbuf);
+        gst_memory_map (mem, &map, GST_MAP_READWRITE);
+        data.data.size = data_size;
+        data.data.data = map.data;
+        nns_logi ("allocated buffer size: %zu", gst_buffer_get_size (*outbuf));
+        gst_memory_unmap (mem, &map);
+        gst_memory_unref (mem);
+        break;
+
+      case _TENSOR_QUERY_CMD_TRANSFER_DATA:
```

Comment:
Please considering streaming data.
I think received data should be handled here..
And sequence of the command buffers are
`_TENSOR_QUERY_CMD_TRANSFER_START | _TENSOR_QUERY_CMD_TRANSFER_DATA | ... | _TENSOR_QUERY_CMD_TRANSFER_DATA | _TENSOR_QUERY_CMD_TRANSFER_END`
The capability of the server does not change. So you don't nned to send a start cmd buf every time.